sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/PushNotificationClient.java

```diff
@@ -351,7 +351,7 @@ private String decryptPayload(String encryptedPayload) throws Throwable {
 
         // Reject the push when computed signature does not match the included signature - it can not be trusted
         throw logger.logExceptionAsError(
-            new RuntimeException("Invalid encrypted push notification payload! Drop the request."));
+            new RuntimeException("Invalid or expired authorization key. Drop the request!"));
```

Comment:
What I meant to suggest with my previous comment was to provide an accurate error message for whatever case we encounter, in other words, we need to tell the user why the payload could not be verified, either because the key is expired or because the payload could not be verified with a valid key. So we need at least two error messages in my opinion. What do you think?

Reply:
Ah. Got it. Here what we are doing is to calculate a signature of the encrypted payload using local stored authKey, and compare the local calculated signature with the one the client received from the server. If does not match, we would think the encrypted payload can not be trusted.

Here, we would always think the local key in client side is valid, and assume the encrypted payload broken or attacked by someone. It should always be "the payload could not be verified with a valid key".

Do you have any suggestion whether we have better error message here? Thanks!

Reply:
I agree that if the payload cannot be verified we should assume the key is valid and the problem relies with the payload. I was thinking that about the scenario where the grace time for renewing the key expires, it doesn't feel right saying that the payload was invalid in such a case, as developers wouldn't know they need to renew the key to attempt the verify operation. So I would either have one distinct error message for each of those two cases, or phrase this as the following:

```suggestion
            new RuntimeException("Invalid encrypted push notification payload or expired authorization key. Dropped the request!"));
```

Also, I'm not sure about using exclamation signs (!) at the end of these messages but honestly that's not a big deal for the time being, in my opinion.

Reply:
For key renewing scenario, it would always be the SDK itself to create a new key for registrar request, then mark the old keys as previousKeys and start the grace time that still try the previousKeys if new key failed in validation.

Reply:
We don't have the case for "expired authorization key" here.

Reply:
Maybe the key itself doesn't expire after the "grace" period passes, but would we allow customers to use a very old key? If not, we need to be clear in letting them know they need to provide a more recent one. If the key is not all that old and the payload cannot be verified, then we let customers know the payload is not trustworthy.

Reply:
Here keys are auto-generated & rotated by SDK (not provided by customer) every 25 hours together with registration renew. Would this be an issue that we should let customer pass in their own keys? If this is OK, we don't have "very old key" scenario. Unable to verify means the payload is not trustworthy.

Reply:
Oh I see, for some reason I thought these keys could come from users. If the SDK is taking care of all this, then probably just telling users the payload could not be verified is enough.